contents/csharp/IntroductionToCSharp.md

```diff
@@ -0,0 +1,246 @@
+
+The usage of C# in game development is fairly common due to the Unity game engine. Unity has great cross-platform compatibility for desktop, web, mobile and console, and has extensive support for 2D/3D games, VR/AR games
+and games that require networking.
+
+## How to Get Started
+
+Getting started with C# is not difficult. You can download Visual Studio and follow the setup instructions for C# programming [here](https://www.guru99.com/download-install-visual-studio.html).
+If you are new to C# but have some familiarity with Java, you may follow the tutorial at [Sololearn](https://www.sololearn.com/Play/CSharp). It is a rather comprehensive tutorial that teaches
+fundamental syntax and concepts in C#. If you feel that certain parts of the tutorial are too simple, you can also skip them.
+
+If you are entirely new to programming, you may find more hands on practice with simpler steps at [CSharp net](https://csharp.net-tutorials.com/getting-started/introduction/). This tutorial tends
+to be more rigorous and goes through in great detail the steps, starting from installing a development environment, to writing basic C# programs and finally topics commonly used in real applications
+, such as file handling and debugging. If you wish to skip certain parts of the tutorial, the structured contents are displayed on the right side of the website.
+
+If you want to have a go at maintaining and enhancing a small project, you may find this [fictitious airline reservation system](http://1000projects.org/airline-reservation-system-a-net-project-with-code.html) project
+to be a good starting point. It covers commonly used components of a software, such as UI (Application User Interface), data storage and handling, and the logic behind the program (such as buying a ticket).
+More similar projects can be found at [1000projects.org](http://1000projects.org/c-projects.html).
+
+If you feel that you have a grasp of C# fundamentals but find it difficult to write programs of bigger scale, such as an Address Book application, you may consult
+[CSharp corner](https://www.c-sharpcorner.com/UploadFile/bd5be5/design-patterns-in-net/) for a list of design patterns that you may employ to better organise and plan your
+program structure. Sometimes, you may find that you have problems collaborating on a C# project. This may be due to some common misconceptions and mistakes you are commiting without
+realisation. You may reduce these problems by reading about [some common mistakes in C# programming](https://www.upwork.com/hiring/development/common-mistakes-in-c-sharp-programming/).
+
+If you want to develop a desktop application for Windows, you may consult [Microsoft's documentation](https://docs.microsoft.com/en-us/visualstudio/get-started/csharp/tutorial-wpf?view=vs-2019)
+on creating an application with the `Windows Presentation Foundation`, a framework that is commonly used for creating UI for Windows Applications that has many useful features for building your UI.
```

Comment:
This section looks good. Perhaps drop the mention of Address Book? That's not supposed to be a big software anyway.